nameko/cli/run.py

```diff
@@ -168,7 +169,14 @@ def main(args):
             import_service(path)
         )
 
-    config = {AMQP_URI_CONFIG_KEY: args.broker}
+    config = {}
+
+    if args.config:
+        with open(args.config) as fle:
+            config = yaml.load(fle)
+    else:
+        config[AMQP_URI_CONFIG_KEY] = args.broker
```

Comment:
I really like the idea of registered config keys, descriptions and defaults. I could then introspect a class and determine what configs I may need to run it and match it up with a preexisting config I already have.

As you said, command line args will end up getting silly, but I did not want to break compatibility. 
One option could be to treat them as deprecated and print out an warning when used.